### ISSUE TITLE
[Desktop Build] Pressing Tab key to navigate pop-up windows options has unreliable behavior.

### ISSUE THREAD

**Issue by RabloGames** [Scirra/Construct-3-bugs#3002]
## Problem description

This problem is on Desktop build only. I've been unable to reproduce on the web-based editor.
When pressing the Tab key to navigate a pop-up window (such as when editing the "add to" action, or functions parameters), the highlighted field sometimes gets completely selected (as with Ctrl-A), but sometimes the cursor is just placed on the left of the field with nothing selected.

As an example, the images below have been screenshotted with the exact same inputs : press enter, then press tab.

![tab](https://user-images.githubusercontent.com/29117306/58745023-1a4a4780-844b-11e9-958f-4e46e3fd6f91.jpg)

## Attach a .c3p

Any "add to instance variable" or "call function" (with parameters) or anything similar will trigger the problem.

## Steps to reproduce

1. Select an action "add to", "subtract from" or anything which can be navigated with tab to highlight a numeric value. You can either double click or press enter, the bug appears anyway.
2. Press tab to select the next field.
3. Repeat around 20 times and observe that **most of the times** the whole field is selected, but not always.

## Observed result

The whole field is not always selected. Sometimes the cursor is simply placed on the left of the field with nothing selected.

## Expected result

The whole field should always be selected.

## More details

**Affected browsers/platforms:** 

Desktop Build (web based editor not affected)

**First affected release:**

r.151

## System details

<details><summary>View details</summary>

Platform information
Browser: NW.js
Browser version: 0.30.0
Browser engine: Chromium
Browser architecture: 64-bit
Context: nwjs
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1.25
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (@004e06da11e10bb08f6632224ae48ee3d5f7b303) NWjs/0.30.0 (like Chrome/66.0.3359.117)
C3 release: r151 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 16 gb
Storage usage (approx): 2.9 gb (18.6%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: no
Supports highp precision: yes
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 970M Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_texture_filter_anisotropic
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 8
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by AshleyScirra** [comment]
Can't reproduce here, I tabbed between the two parameters of Ghost Shooter's set angle toward mouse action repeatedly dozens of time and it worked fine every time. Also the code is identical between Chrome and the desktop version, so it's weird if it only happens in one. Unfortunately if we cannot observe the issue there's little that can be done about it.

**Comment by AshleyScirra** [comment]
I made a shotgun change for the next release anyway which might help, but no idea if it will since it appeared to work fine for me.

**Comment by RabloGames** [comment]
Thanks to have looked into this. 
To give you further details, I tried switching between events of a "set angle towards position" event, and the bug never occured.
However, from a "subtract from instance variable" action, it occurs frequently in my project.
I tried creating a new project but it never happen in the new project. So I copied a simple "subtract from" event 1000 times and now the problem occurs **every time**! (on a subtract from action, it still never occurs on "set angle towards position")

Here is my simple project 
[tab_select_bug.zip](https://github.com/Scirra/Construct-3-bugs/files/3269947/tab_select_bug.zip)

It also seem to occur less frequently/never if you keep pressing tab in the action. Just press enter on an existing action, then press tab, see if it happened, then press ESC if it didn't.
